### PR TITLE
ADD: [droid;aml+rk] specific aml and rk egl windowing for droid; allows framerate autoswitch

### PR DESCRIPTION
**Pull request by koying**
Added bonus: aml return hdmi size rather than fb one, so no more "why is my box in 720p" ;)

I double 24p, returning 48fps to Kodi to avoid waiting in loop.
I'm not so sure it makes sense, really...

/cc @FernetMenta @fritsch @Montellese @davilla

### CODE REVIEW

File: xbmc/cores/VideoRenderers/BaseRenderer.cpp [hunk 1]
```diff
@@ -225,7 +225,7 @@ RESOLUTION CBaseRenderer::FindClosestResolution(float fps, float multiplier, RES
 
   float fRefreshRate = fps;
 
-  float last_diff = fRefreshRate;
+  float last_diff = fRefreshRate*2;
```
Comment (FernetMenta) [inline xbmc/cores/VideoRenderers/BaseRenderer.cpp]: this changes behavior for all platforms. why is this hidden in this commit? why did you change it?

**Comment by koying** [comment]
Refactored.

jenkins build this please

File: xbmc/windowing/egl/EGLWrapper.cpp [hunk 2]
```diff
@@ -22,6 +22,8 @@
 #ifdef HAS_EGL
 #include "utils/log.h"
 #include "EGLNativeTypeAndroid.h"
+#include "EGLNativeTypeAmlAndroid.h"
+#include "EGLNativeTypeRKAndroid.h"
```
Comment (wsnipex) [inline xbmc/windowing/egl/EGLWrapper.cpp]: Does it make sense to build those on all platforms that have EGL?

Comment (wsnipex) [inline xbmc/windowing/egl/EGLWrapper.cpp]: actually, that comment was meant for the Makefile below.

Comment (koying) [inline xbmc/windowing/egl/EGLWrapper.cpp]: Well, no more, no less than EGLNativeTypeAndroid, really. Just following the convention of no ifdef here, apparently

Comment (koying) [inline xbmc/windowing/egl/EGLWrapper.cpp]: Same response ;)

**Comment by koying** [comment]
@FernetMenta @wsnipex @Montellese Still comments?

File: xbmc/windowing/egl/EGLWrapper.cpp (hunk 2 above)
Comment (FernetMenta) [inline xbmc/windowing/egl/EGLWrapper.cpp]: sorry, I haven't noticed the ping.
I think we should not continue this very weird style of including everything and ifdeffing each implementation of the methods in the body. I see no reason why we should compile and include RPi specific files in Android build and vice versa.

Comment (koying) [inline xbmc/windowing/egl/EGLWrapper.cpp]: Totally agree. Will do a "pre-commit" with proper #ifdefs. No point to have Wayland for Android, either ;)

**Comment by koying** [comment]
jenkins build this please

File: xbmc/windowing/egl/EGLNativeTypeAmlAndroid.cpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "system.h"
+#include <EGL/egl.h>
+#include "EGLNativeTypeAmlAndroid.h"
+#include "utils/log.h"
+#include "guilib/gui3d.h"
+#if defined(TARGET_ANDROID)
```
Comment (wsnipex) [inline xbmc/windowing/egl/EGLNativeTypeAmlAndroid.cpp]: not needed anymore

Comment (koying) [inline xbmc/windowing/egl/EGLNativeTypeAmlAndroid.cpp]: Yep. Will go through all of them.

File: xbmc/windowing/egl/EGLNativeTypeRKAndroid.cpp [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+ */
+
+#include <stdlib.h>
+
+#include "system.h"
+#include <EGL/egl.h>
+#include "EGLNativeTypeRKAndroid.h"
+#include "utils/log.h"
+#include "guilib/gui3d.h"
+#if defined(TARGET_ANDROID)
```
Comment (wsnipex) [inline xbmc/windowing/egl/EGLNativeTypeRKAndroid.cpp]: same

**Comment by koying** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
thanks, IMO much better now

**Comment by koying** [comment]
jenkins build and merge
